### ISSUE TITLE
Question: is it possible to include local vars/calls in structure?

### ISSUE THREAD

**Issue by garnett** [jpsim/SourceKitten#151]
consider snippet:

```swift
private func foo(dictionary: [Int: Int]) -> Int {
    let a = 2
    dictionary.flatMap { foo in return foo }
    return 1
}
```

SourceKitten Produces:

```
  "key.substructure" : [
    {
      "key.kind" : "source.lang.swift.decl.function.free",
      "key.offset" : 22,
      "key.nameoffset" : 27,
      "key.namelength" : 27,
      "key.bodyoffset" : 63,
      "key.bodylength" : 73,
      "key.accessibility" : "source.lang.swift.accessibility.private",
      "key.substructure" : [
        {
          "key.kind" : "source.lang.swift.decl.var.parameter",
          "key.offset" : 31,
          "key.nameoffset" : 0,
          "key.namelength" : 0,
          "key.length" : 10,
          "key.typename" : "[Int: Int]",
          "key.name" : "dictionary"
        },
        {
          "key.kind" : "source.lang.swift.decl.var.parameter",
          "key.offset" : 103,
          "key.nameoffset" : 0,
          "key.namelength" : 0,
          "key.length" : 3,
          "key.typename" : "foo",
          "key.name" : "foo"
        }
      ],
      "key.name" : "foo(_:)",
      "key.length" : 115
    }
  ],
  "key.offset" : 0,
  "key.diagnostic_stage" : "source.diagnostic.stage.swift.parse",
  "key.length" : 138
}
```

I've expected to see `a` here or at least `expr.call` for `flatMap`. However it might be that structure is  just *structure*: but then I do not realize how to distinct between function parameters and `foo` parameter in closure (needed for https://github.com/realm/SwiftLint/issues/415)

Non related:
I wanted to debug it myself, but 
```
xcrun sourcekitd-test
xcrun: error: unable to find utility "sourcekitd-test", not a developer tool or in PATH
```
Any advice on that?

**Comment by jpsim** [comment]
SourceKitten's `Structure` builder is really just a thin wrapper around SourceKit's `editor.open` request. See [Structure.swift](https://github.com/jpsim/SourceKitten/blob/master/Source/SourceKittenFramework/Structure.swift) and [`Request.EditorOpen`](https://github.com/jpsim/SourceKitten/blob/0.8.0/Source/SourceKittenFramework/Request.swift#L128-L141) which then is [sent](https://github.com/jpsim/SourceKitten/blob/0.8.0/Source/SourceKittenFramework/Request.swift#L205-L209) and the response from SourceKit is mapped to a `Dictionary<String, SourceKitRepresentable>`, then finally converted to JSON and printed.

So in other words, to expand the contents of `Structure`, you'd need to do that in [SourceKit](https://github.com/apple/swift/tree/master/tools/SourceKit) itself.

`sourcekitd-test` is a tool built to test SourceKit that was originally included in some early Xcode 6 betas, but subsequently removed, so it hasn't shipped with Xcode since Xcode 6 beta 3-4 IIRC. You can find the source for the tool here: https://github.com/apple/swift/tree/master/tools/SourceKit/tools/sourcekitd-test

I hope that helps!

**Comment by garnett** [comment]
Since it works as designed I'm closing the issue
